mantidproject/mantid
# Mask zero counting detectors in DirectILLIntegrateVanadium

**Description of work.**

This adds a zero masking step to the `DirectILLIntegrateVanadium` algorithm. The purpose is to prevent infinities when the sample spectra is divided by the integrated vanadium.

**Report to:** Jacques Ollivier/ILL

**To test:**

Check that script below runs. Try setting different spectra to zero and see that they get masked. The script uses data from the unit test data directory so be sure to set the data search paths correct.

```python
DirectILLCollectData('ILL/IN4/084446.nxs', OutputWorkspace='vana', OutputEPPWorkspace='epps')
ws = mtd['vana']
ws.dataY(9).fill(0.)
intgrl = DirectILLIntegrateVanadium('vana', EPPWorkspace='epps')
spectrumInfo = intgrl.spectrumInfo()
assert(spectrumInfo.isMasked(9))
```

Fixes #23580.

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
